core/templates/components/entity-creation-services/exploration-creation.service.spec.ts

```diff
@@ -0,0 +1,224 @@
+import { CsrfTokenService } from 'services/csrf-token.service';
+import { LoaderService } from 'services/loader.service';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { ExplorationCreationBackendApiService, ExplorationCreationResponse } from './exploration-creation-backend-api.service';
+import { ExplorationCreationService } from './exploration-creation.service';
+
+class MockExploratinoCreationBackendApiService {
+  throwError: boolean = false;
+  message: string;
+  registerNewExplorationAsync(): object {
+    return {
+      then: (
+          successCallback: (response: {explorationId: string}) => void,
+          errorCallback: (errorMessage: string) => void
+      ) => {
+        if (this.throwError) {
+          errorCallback(this.message);
+        } else {
+          successCallback({
+            explorationId: 'exp1'
+          });
+        }
+      }
+    };
+  }
+}
+
+class MockWindowRef {
+  _window = {
+    location: {
+      _href: '',
+      get href(): string {
+        return this._href;
+      },
+      set href(val) {
+        this._href = val;
+      }
+    }
+  };
+  get nativeWindow() {
+    return this._window;
+  }
+}
+
+describe('ExplorationCreationService', () => {
+  let ecs: ExplorationCreationService;
+  let ecbas: MockExploratinoCreationBackendApiService;
+  let loaderService: LoaderService;
+  let siteAnalyticsService: SiteAnalyticsService;
+  let urlInterpolationService: UrlInterpolationService;
+  let csrfTokenService: CsrfTokenService;
+  let alertsService: AlertsService;
+  let windowRef: MockWindowRef;
+  let ngbModal: NgbModal;
+
+  beforeEach(() => {
+    windowRef = new MockWindowRef();
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [
+        {
+          provide: ExplorationCreationBackendApiService,
+          useClass: MockExploratinoCreationBackendApiService
+        },
+        {
+          provide: WindowRef,
+          useValue: windowRef
+        }
+      ]
+    });
+
+    ecs = TestBed.inject(ExplorationCreationService);
+    ecbas = (TestBed.inject(
+      ExplorationCreationBackendApiService) as unknown) as
+      MockExploratinoCreationBackendApiService;
+    loaderService = TestBed.inject(LoaderService);
+    siteAnalyticsService = TestBed.inject(SiteAnalyticsService);
+    urlInterpolationService = TestBed.inject(UrlInterpolationService);
+    csrfTokenService = TestBed.inject(CsrfTokenService);
+    alertsService = TestBed.inject(AlertsService);
+    ngbModal = TestBed.inject(NgbModal);
+  });
+
+  it('should not create a new exploration if another exploration' +
+    ' creation is in progress', () => {
+    spyOn(ecbas, 'registerNewExplorationAsync');
+    ecs.explorationCreationInProgress = true;
+
+    expect(ecs.createNewExploration()).toBe(undefined);
+    expect(ecbas.registerNewExplorationAsync).not.toHaveBeenCalled();
+  });
+
+  it('should show \'Creating exploration\' loading screen while' +
```

Comment:
Service doesn't show anything, should we make it `should change loadingMessage to Creating exploration`

Also, can we have this structure:
```
describe: ExplorationCreationService
    describe: on calling createNewExploration
       it: should not create a new exploration if another exploration creation is in progress
       it: should change loadingMessage to Creating exploration while creating exploration
       it: ......
```

Reply:
Done